pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -420,6 +439,42 @@ func removeFinalizerFromCRDInstance(ctx context.Context, instance *cnsnodevmatta
 	}
 }
 
+// isVmCrPresent checks whether VM CR is present in SV namespace
+// with given vmuuid.
+func isVmCrPresent(ctx context.Context, vmuuid string, namespace string) (bool, error) {
+	log := logger.GetLogger(ctx)
+	restConfig, err := k8sclientconfig.GetConfig()
+	if err != nil {
+		log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return false, err
+	}
+	vmOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig,
+		vmoperatortypes.GroupName)
+	if err != nil {
+		log.Errorf("failed to create vmOperatorClient. Error: %+v", err)
+		return false, err
+	}
```

Comment:
Why do we have to create a new client each time this method is executed? Can we not reuse or re-establish the clients?

Reply:
Addressed in latest commit.